app/domain/authentication/authn_iam/authenticator.rb

```diff
@@ -6,72 +6,76 @@ module Authentication
 
-        response_hash = identity_hash(response_from_signed_request(signed_aws_headers))
-        trusted = response_hash != false
-
-        trusted && iam_role_matches?(input.username, response_hash)
-      end
-
-      def identity_hash(response)
-        Rails.logger.debug(
-          LogMessages::Authentication::AuthnIam::GetCallerIdentityBody.new(
-            response.body
-          )
+        iam_role_matches?(
+          login: input.username,
+          aws_arn: aws_response[:arn],
+          aws_account: aws_response[:account]
         )
-
-        if response.code < 300
-          Hash.from_xml(response.body)
-        else
-          Rails.logger.error(
-            Errors::Authentication::AuthnIam::IdentityVerificationErrorCode.new(
-              response.code
-            )
-          )
-          false
-        end
       end
 
-      def iam_role_matches?(login, response_hash)
-        split_assumed_role = response_hash["GetCallerIdentityResponse"]["GetCallerIdentityResult"]["Arn"].split(":")
+      private
 
+      # Parses STS Caller Identity response for the Conjur host and resolves it against the provided login
+      def iam_role_matches?(login:, aws_arn:, aws_account:)
         # removes the last 2 parts of login to be substituted by the info from getCallerIdentity
-        host_prefix = (login.split("/")[0..-3]).join("/")
-        aws_role_name = split_assumed_role[5].split("/")[1]
-        aws_account_id = response_hash["GetCallerIdentityResponse"]["GetCallerIdentityResult"]["Account"]
-        aws_user_id = response_hash["GetCallerIdentityResponse"]["GetCallerIdentityResult"]["UserId"]
-        host_to_match = "#{host_prefix}/#{aws_account_id}/#{aws_role_name}"
+        host_prefix = (login.split('/')[0..-3]).join('/')
 
-        Rails.logger.debug(
+        arn_parts = aws_arn.split(':')
+        aws_role_name = arn_parts[5].split('/')[1]
+        host_to_match = "#{host_prefix}/#{aws_account}/#{aws_role_name}"
+
+        @logger.debug(
           LogMessages::Authentication::AuthnIam::AttemptToMatchHost.new(
-            aws_user_id,
+            login,
             host_to_match
           )
         )
 
         login.eql?(host_to_match)
       end
 
-      def aws_signed_url
-        return 'https://sts.amazonaws.com/?Action=GetCallerIdentity&Version=2011-06-15'      
+      def extract_relevant_data(response)
+        {
+          arn: response.dig('GetCallerIdentityResponse', 'GetCallerIdentityResult', 'Arn'),
+          account: response.dig('GetCallerIdentityResponse', 'GetCallerIdentityResult', 'Account')
+        }
       end
 
-      def response_from_signed_request(aws_headers)
-        Rails.logger.debug(LogMessages::Authentication::AuthnIam::RetrieveIamIdentity.new)
+      # Call to AWS STS endpoint using the provided authentication header
+      def attempt_signed_request(signed_headers)
+        aws_request = URI("https://#{signed_headers['host']}/?Action=GetCallerIdentity&Version=2011-06-15")
```

Comment:
Hey @jvanderhoof , could we consider using `sts_host` as the key here, to disambiguate it from Conjur "host"?